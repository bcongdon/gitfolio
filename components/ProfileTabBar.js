import React from "react";

import { Tab, Tabs } from "native-base";

/*
  ProfileTabBar is the component for the bar of tabs at the bottom of the user profile view
 */
export default class ProfileTabBar extends React.Component {
  render() {
    return (
      <Tabs
        tabBarPosition={"top"}
        onChangeTab={this.props.onChangeTab}
        page={this.props.activeTab}
      >
        {/* Tab for the user profile view */}
        <Tab heading={"Profile"}>{this.props.profileView}</Tab>

        {/* Tab for the repositories */}
        <Tab heading={"Repos"}>{this.props.repositoriesView}</Tab>

        {/* Tab for the "User Following" view */}
        <Tab heading={"Following"}>{this.props.followingView}</Tab>

        {/* Tab for the "User Followers" view */}
        <Tab heading={"Followers"}>{this.props.followersView}</Tab>
      </Tabs>
    );
  }
}
