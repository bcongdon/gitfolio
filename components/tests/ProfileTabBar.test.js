import ProfileTabBar from "../ProfileTabBar";
import React from "react";

import renderer from "react-test-renderer";

it("renders without crashing", () => {
  let component = <ProfileTabBar navigationState={{}} navigation={{}} />;

  const rendered = renderer.create(component).toJSON();
  expect(rendered).toBeTruthy();
});

it("should pass snapshot test", () => {
  let component = <ProfileTabBar navigationState={{}} navigation={{}} />;

  const rendered = renderer.create(component).toJSON();
  expect(rendered).toMatchSnapshot();
});
