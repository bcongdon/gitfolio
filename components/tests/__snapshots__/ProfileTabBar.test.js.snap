// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should pass snapshot test 1`] = `
<View
  onLayout={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "borderBottomWidth": 0.5,
          "borderColor": "#a7a6ab",
          "elevation": 3,
          "flexDirection": "row",
          "height": 50,
          "justifyContent": "space-around",
          "shadowColor": undefined,
          "shadowOffset": undefined,
          "shadowOpacity": undefined,
          "shadowRadius": undefined,
        },
        Object {
          "backgroundColor": "#F8F8F8",
        },
        Object {},
      ]
    }
    virtual={undefined}
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      collapsable={undefined}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "flex": 1,
          "opacity": 1,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <View
        active={true}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#F8F8F8",
            "flex": 1,
            "flexDirection": "row",
            "justifyContent": "center",
          }
        }
        virtual={undefined}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#007aff",
              "fontFamily": "System",
              "fontSize": 16,
              "fontWeight": "600",
              "marginLeft": 7,
              "marginRight": 7,
            }
          }
          uppercase={false}
          virtual={undefined}
        >
          Profile
        </Text>
      </View>
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      collapsable={undefined}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "flex": 1,
          "opacity": 1,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <View
        active={false}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#F8F8F8",
            "flex": 1,
            "flexDirection": "row",
            "justifyContent": "center",
          }
        }
        virtual={undefined}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#6b6b6b",
              "fontFamily": "System",
              "fontSize": 16,
              "marginLeft": 7,
              "marginRight": 7,
            }
          }
          uppercase={false}
          virtual={undefined}
        >
          Repos
        </Text>
      </View>
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      collapsable={undefined}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "flex": 1,
          "opacity": 1,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <View
        active={false}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#F8F8F8",
            "flex": 1,
            "flexDirection": "row",
            "justifyContent": "center",
          }
        }
        virtual={undefined}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#6b6b6b",
              "fontFamily": "System",
              "fontSize": 16,
              "marginLeft": 7,
              "marginRight": 7,
            }
          }
          uppercase={false}
          virtual={undefined}
        >
          Following
        </Text>
      </View>
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      collapsable={undefined}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "flex": 1,
          "opacity": 1,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <View
        active={false}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#F8F8F8",
            "flex": 1,
            "flexDirection": "row",
            "justifyContent": "center",
          }
        }
        virtual={undefined}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#6b6b6b",
              "fontFamily": "System",
              "fontSize": 16,
              "marginLeft": 7,
              "marginRight": 7,
            }
          }
          uppercase={false}
          virtual={undefined}
        >
          Followers
        </Text>
      </View>
    </View>
    <View
      collapsable={undefined}
      style={
        Object {
          "backgroundColor": "#007aff",
          "bottom": 0,
          "height": 4,
          "left": 0,
          "position": "absolute",
          "width": 187.5,
        }
      }
    />
  </View>
  <RCTScrollView
    alwaysBounceVertical={false}
    automaticallyAdjustContentInsets={false}
    contentOffset={
      Object {
        "x": 0,
      }
    }
    directionalLockEnabled={true}
    horizontal={true}
    keyboardDismissMode="on-drag"
    keyboardShouldPersistTaps="handled"
    onMomentumScrollBegin={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    pagingEnabled={true}
    scrollEnabled={true}
    scrollEventThrottle={16}
    scrollsToTop={false}
    showsHorizontalScrollIndicator={false}
  >
    <View>
      <View
        style={
          Object {
            "width": 750,
          }
        }
      >
        <View
          heading="Profile"
          style={
            Object {
              "backgroundColor": "#FFF",
              "flex": 1,
            }
          }
          virtual={undefined}
        />
      </View>
      <View
        style={
          Object {
            "width": 750,
          }
        }
      >
        <View
          heading="Repos"
        />
      </View>
      <View
        style={
          Object {
            "width": 750,
          }
        }
      >
        <View
          heading="Following"
        />
      </View>
      <View
        style={
          Object {
            "width": 750,
          }
        }
      >
        <View
          heading="Followers"
        />
      </View>
    </View>
  </RCTScrollView>
</View>
`;
